File: practice-1/exercise-1/check-balance.ts
import {clusterApiUrl, Connection, LAMPORTS_PER_SOL, PublicKey} from "@solana/web3.js"

const connection = new Connection(clusterApiUrl("devnet"))
console.log("Connected to devnet!")

const publicKey = new PublicKey("GLFNFaEujqFTmVQvCJ2bXRdpvBkDtpvSjAZxtY2XAReN");
const balanceInLamports = await connection.getBalance(publicKey)

const balanceInSOL = balanceInLamports / LAMPORTS_PER_SOL

console.log(
  `💰 The balance for the wallet at address ${publicKey} is: ${balanceInSOL}`
);

File: practice-1/rust-grinder/src/bin/check_balance.rs
use std::str::FromStr;
use solana_client::rpc_client::RpcClient;
use solana_sdk::native_token::LAMPORTS_PER_SOL;
use solana_sdk::pubkey::Pubkey;

fn main() {
    let conn = RpcClient::new("https://api.devnet.solana.com");

    let public_key = Pubkey::from_str("FKV9HNf8uPnMYGYVv2zAeLf3KDugtChwbg3oZrAXYcdo").expect("invalid pubkey");

    let account_balance = conn.get_balance(&public_key).expect("failed to get balance");

    println!("Balance: {}", account_balance / LAMPORTS_PER_SOL);
}